test/integration/integration_test.h

```diff
@@ -4,24 +4,25 @@
 
 #include "gtest/gtest.h"
 
-class IntegrationTest : public BaseIntegrationTest, public testing::Test {
+class IntegrationTest : public BaseIntegrationTest,
+                        public testing::TestWithParam<Network::Address::IpVersion> {
 public:
   /**
-   * Global initializer for all integration tests.
+   * Initializer for individual integration tests.
    */
-  static void SetUpTestCase() {
+  void SetUp() override {
```

Comment:
I think there's a little more to it than just switching `SetUpTestCase` with `SetUp` when we want to make test setup/teardown local to each test and remove interference between tests. Take a look at https://github.com/lyft/envoy/blob/master/test/integration/integration.h#L169 and all the `static` fields. Some (all?) of these should probably become instance variables.